File: week4.2/anagram.py
def is_anagram(s, t):
    frequency_s = {}
    frequency_t = {}
    for i in s:
        if i in frequency_s:
            frequency_s[i] += 1
        else:
            frequency_s[i] = 1
    for i in t:
        if i in frequency_t:
            frequency_t[i] += 1
        else:
            frequency_t[i] = 1
    if frequency_s == frequency_t:
        return True
    else:
        return False
print(is_anagram("anagram", "nagaram"))

File: week4.2/anagram.js
var isAnagram = function(s, t) {
    const frequency1 = {}
    const frequency2 = {}
    for(let ele of s){
        if(frequency1[ele]){
            frequency1[ele]+=1
        }else{
            frequency1[ele]=1
        }
    }
    for(let ele of t){
        if(frequency2[ele]){
            frequency2[ele]+=1
        }else{
            frequency2[ele]=1
        }
    }
    for(let key in frequency1){
        if(frequency1[key]!==frequency2[key]){
            return false
        }
    }
    return true
   
};

console.log(isAnagram('aacc','ccac'))
console.log(isAnagram("anagram", "nagaram"))